interfaces/builtin/download.go

```diff
@@ -0,0 +1,325 @@
+     interface="org.freedesktop.DBus.Introspectable"
+     path=/
+     member=Introspect
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     interface="org.freedesktop.DBus.Introspectable"
+     path=/com/canonical/applications/download/**
+     member=Introspect
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Allow DownloadManager to send us signals, etc
+dbus (receive)
+     bus=session
+     interface=com.canonical.applications.Download{,er}Manager
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/applications/download/@{SNAP_LABEL_DBUS}/**
+     interface=com.canonical.applications.Download
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive)
+     bus=session
+     path=/com/canonical/applications/download/@{SNAP_LABEL_DBUS}/**
+     interface=org.freedesktop.DBus.Properties
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/applications/download/@{SNAP_LABEL_DBUS}/**
+     interface=com.canonical.applications.GroupDownload
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Be explicit about the allowed members we can send to
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownload
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=createDownloadGroup
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (receive, send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloads
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=getAllDownloadsWithMetadata
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=defaultThrottle
+     peer=(label=###SLOT_SECURITY_TAGS###),
+dbus (send)
+     bus=session
+     path=/
+     interface=com.canonical.applications.DownloadManager
+     member=isGSMDownloadAllowed
+     peer=(label=###SLOT_SECURITY_TAGS###),
+# Explicitly deny DownloadManager APIs apps shouldn't have access to in order
+# to make sure they aren't accidentally added in the future (see LP: #1277578
+# for details)
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=allowGSMDownload,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=createMmsDownload,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=exit,
+audit deny dbus (send)
+     bus=session
+     interface=com.canonical.applications.DownloadManager
+     member=setDefaultThrottle,
+`)
```

Comment:
The way snappy stitches together, explicit deny rules are problematic. Please instead remove these explicit denies and simply document that allowGSMDownload, createMmsDownload, exit and setDefaultThrottle are intentionally left out of downloadConnectedPlugAppArmor since they are privileged. Perhaps we can use an attribute or new interface (eg, download-manager-control) when snaps need access to these privileged methods.